### ISSUE TITLE
Svgs have a fixed size without veiwbox property set

### ISSUE THREAD

**Issue by I-am-tk** [marella/material-design-icons#6]
@marella Thanks for the great library.

While I was using it, I noticed the size of the icons is not changing if the width and height are set to it. All the SVGs do not have a `viewbox` property. 

It will be great if you add this.

**Comment by marella** [comment]
Hi, I'm seeing that `viewBox` property is set in the SVGs. Can you please give some icon names that are missing it.

If this issue is also present in the SVGs downloaded from the [fonts site](https://fonts.google.com/icons), it is better to report it in the [official repo](https://github.com/google/material-design-icons/issues) to fix it at the source.

**Comment by marella** [comment]
Feel free to re-open if you are still facing this issue.

**Comment by marella** [comment]
Looks like this issue is present in Material Symbols but not Material Icons. See marella/material-symbols#5.

**Comment by adamreisnz** [comment]
I am facing this issue with this repository. None of the icons seem to have a `viewbox` property added to them:

<img width="615" alt="image" src="https://github.com/marella/material-design-icons/assets/490562/4c056806-042b-4cf8-b937-fec830f609a0">

Is there an easy way to add it to each component post-import?

**Comment by ratijas** [comment]
tl;dr: SVGO pre-processor used to remove viewbox by default[^1]. Should not be doing that anymore since SVGO 4.0 onward[^2], but for the time being your installation might still be using an older version and needs `removeViewBox: false` plugin parameter in _vite.config.js_.

```js
import { defineConfig } from "vite";
import vue from "@vitejs/plugin-vue";
import svgLoader from "vite-svg-loader";

// https://vite.dev/config/
export default defineConfig({
  plugins: [
    vue(),
    svgLoader({
      svgoConfig: {
        multipass: true,
        plugins: [
          {
            name: "preset-default",
            params: {
              overrides: {
                // @see https://github.com/svg/svgo/issues/1128
                removeViewBox: false,
              },
            },
          },
        ],
      },
    }),
  ],
});

```

---

I'm facing the same issue. Using vite + Vue-ts.

As per README, I
- added package `"@material-design-icons/svg": "^0.14.15"`
- set up [vite-svg-loader](https://github.com/jpkleemans/vite-svg-loader) instead of the suggested [@svgv/webpack](https://www.npmjs.com/package/@svgv/webpack) because I'm using vite instead of webpack
- `import Face from '@material-design-icons/svg/filled/face.svg';`
- `<Face class="app" />`
- ```css
  .app {
    width: 1em;
    height: 1em;
    border: 1px solid red;
  }
  ```

And here is what I'm getting:

<img width="86" height="72" alt="Image" src="https://github.com/user-attachments/assets/748a2b9d-3281-496b-9875-3ee216bd2505" />

```html
<svg data-v-7a7a37b1="" xmlns="http://www.w3.org/2000/svg" width="24" height="24" class="app">
<path d="M9 11.75a1.25 1.25 0 1 0 0 2.5 1.25 1.25 0 0 0 0-2.5m6 0a1.25 1.25 0 1 0 0 2.5 1.25 1.25 0 0 0 0-2.5M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8 0-.29.02-.58.05-.86 2.36-1.05 4.23-2.98 5.21-5.37a9.97 9.97 0 0 0 10.41 3.97c.21.71.33 1.47.33 2.26 0 4.41-3.59 8-8 8"></path>
</svg>
```

note how there is width and height attributes set, but viewBox is missing, resulting in a cropped unscalable image.

---

Browsing through vite-svg-loader repo, I found this thread[^1], which points at SVGO pre-processor being a culprit: its `removeViewBox` parameter which should be set to false in the plugin config, and which has been recently dropped from the default preset (but at the time of writing vite-svg-loader is yet to upgrade[^3]).

[^1]: https://github.com/jpkleemans/vite-svg-loader/issues/123
[^2]: https://github.com/svg/svgo/releases/tag/v4.0.0
[^3]: https://github.com/jpkleemans/vite-svg-loader/blob/e0ef76454bb3ede2c5a5815943568decdb732b34/package.json#L15
